### PR TITLE
Allow to upload files with unlimited size through the user's Arweave wallet

### PR DESCRIPTION
**Pull request by JuanRdBO**
Screenshot from 2021-10-26 00-41-21

### CODE REVIEW

**Comment by JuanRdBO** [comment]
Hey @bhgames I'd appreciate a review of this! :-)

File: js/packages/web/src/actions/nft.tsx [hunk 1]
```diff
@@ -42,7 +47,109 @@ interface IArweaveResult {
...
+
+    const fileType = filesMoreThan10mb[i].type || "glb";
+
+    console.log("Uploading", filesMoreThan10mb[i].name)
+
+    const arweaveTransacionId = await arweaveUpload(await filesMoreThan10mb[i].arrayBuffer(), arWallet, fileType)
+
+    //! ARWEAVE
+
+    if (fileType == "image/jpeg") {
```
Comment (bhgames) [inline js/packages/web/src/actions/nft.tsx:120]: Check image/gif too, these also work

File: js/packages/web/src/actions/nft.tsx [hunk 2]
```diff
@@ -42,7 +47,109 @@ interface IArweaveResult {
...
+  }
+
+  // upload manifest file
+  const manifestTxId = await arweaveUpload(JSON.stringify(metadataContent), arWallet, "application/json", mintKey)
+
+  return manifestTxId
+}
+
 const uploadToArweave = async (data: FormData): Promise<IArweaveResult> => {
+  console.log("ARWEAVEDATA", data)
```
Comment (bhgames) [inline js/packages/web/src/actions/nft.tsx:152]: r3emove

File: js/packages/web/src/actions/nft.tsx [hunk 3]
```diff
@@ -113,8 +221,18 @@ export const mintNFT = async (
     },
   };
 
+
+  //! UPLOADER
+
+  const filesLessThan10mb = files.filter((m) => m.size < 10_000_000);
```
Comment (bhgames) [inline js/packages/web/src/actions/nft.tsx:227]: 9mb is safer limit

File: js/packages/web/src/actions/nft.tsx [hunk 4]
```diff
@@ -220,35 +338,48 @@ export const mintNFT = async (
...
-    (acc: Record<string, Array<{ name: string; value: string }>>, f) => {
-      acc[f.name] = [{ name: 'mint', value: mintKey }];
-      return acc;
-    },
-    {},
-  );
-  data.append('tags', JSON.stringify(tags));
-  realFiles.map(f => data.append('file[]', f));
+    // TODO: convert to absolute file name for image
+    console.log("PREUPLOAD AR", data, txid, env)
```
Comment (bhgames) [inline js/packages/web/src/actions/nft.tsx:359]: remove

File: js/packages/web/src/actions/nft.tsx [hunk 5]
```diff
@@ -220,35 +338,48 @@ export const mintNFT = async (
...
+      m => m.filename === RESERVED_TXN_MANIFEST,
+    );
+    
+    arweaveTxId = metadataFile?.transactionId!
 
-  const result: IArweaveResult = await uploadToArweave(data);
-  progressCallback(6);
+  } else {
+    if (!arWallet) throw new Error("Detected files with more than 10mb but no wallet provided");
+    progressCallback(6);
```
Comment (bhgames) [inline js/packages/web/src/actions/nft.tsx:371]: same callback again - 6

**Review by bhgames — CHANGES_REQUESTED** [review]

**Comment by bhgames** [comment]
Closing due to inactivity

**Pull request closed without merging**